synchronization/trigger.go

```diff
@@ -13,41 +13,82 @@ import (
 	"time"
 )
 
+// proxy interface for time.Ticker
+type Ticker interface {
+	// The channel on which the ticks are delivered.
+	C() <-chan time.Time
+	// Stop turns off a ticker. After Stop, no more ticks will be sent.
+	// Stop does not close the channel, to prevent a concurrent goroutine
+	// reading from the channel from seeing an erroneous "tick".
+	Stop()
+}
+
+type timeTicker struct {
+	time.Ticker
+}
+
+func NewTimeTicker(d time.Duration) Ticker {
+	return &timeTicker{*time.NewTicker(d)}
+}
+
+func (t *timeTicker) C() <-chan time.Time {
+	return t.Ticker.C
+}
+
+type HookTicker struct {
+	C_    func() <-chan time.Time
+	Stop_ func()
+}
+
+func (t *HookTicker) C() <-chan time.Time {
+	return t.C_()
+}
+
+func (t *HookTicker) Stop() {
+	t.Stop_()
+}
+
+var noopChannel = make(chan time.Time)
+
+func NewHookTicker() *HookTicker {
```

Comment:
This was mentioned in other comments but i'll summarise here - maybe consider simplifying the usage of `HookTicker` by:
1. Having `HookTicker` manage the trigger channel internally (not exposed)
2. Add a `tick()` (writes to the channel)
3. Add an `isStopped` property (set by `.Stop()`), and an `IsStopped()` getter. 
I think it would make `HookTicker` just as powerful, but would make the usage much simpler and clearer.

Reply:
see the new `TestTicker`